src/core/Akka.Streams.TestKit/SubscriberFluentBuilder.cs

```diff
@@ -5,647 +5,590 @@
+        ///     test.Tell("ping");
+        ///     return ExpectMsg<string>();
+        /// });
+        /// ]]>
+        /// </summary>
+        public async Task<TOther> WithinAsync<TOther>(TimeSpan max, Func<TOther> execute, CancellationToken cancellationToken = default)
         {
-            ExpectNextUnorderedTask(Probe, timeout, elems)
-                .ConfigureAwait(false).GetAwaiter().GetResult();
-            return this;
+            await ExecuteAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
+            return await Probe.TestProbe.WithinAsync(max, execute, cancellationToken: cancellationToken)
+                .ConfigureAwait(false);
         }
         
-        /// <inheritdoc cref="TestSubscriber.ManualProbe{T}.ExpectNextUnorderedAsync(Nullable{TimeSpan},T[])"/>
-        public SubscriberFluentBuilder<T> ExpectNextUnorderedAsync(TimeSpan? timeout,params T[] elems)
+        public async Task WithinAsync(
+            TimeSpan max,
+            Func<Task> actionAsync,
+            TimeSpan? epsilonValue = null,
+            CancellationToken cancellationToken = default)
         {
-            Append(ExpectNextUnorderedTask(Probe, timeout, elems));
-            return this;
+            await ExecuteAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
+            await Probe.TestProbe.WithinAsync(max, actionAsync, epsilonValue, cancellationToken)
+                .ConfigureAwait(false);
         }
         
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static async Task ExpectNextUnorderedTask(ManualProbe<T> probe, TimeSpan? timeout, params T[] elems)
+        /// <summary>
+        /// Attempt to drain the stream into an IAsyncEnumerable (by requesting long.MaxValue elements).
+        /// </summary>
+        public async IAsyncEnumerable<T> ToStrictAsync(
+            TimeSpan atMost,
+            [EnumeratorCancellation] CancellationToken cancellationToken = default)
         {
-            var len = elems.Length;
-            var e = await probe.ExpectNextNAsync(len, timeout)
-                .ToListAsync().ConfigureAwait(false);
-            AssertEquals(e.Count, len, "expected to get {0} events, but got {1}", len, e.Count);
+            await ExecuteAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
+            await foreach (var item in ManualProbe<T>.ToStrictTask(Probe, atMost, cancellationToken))
+            {
+                yield return item;
+            }
+        }
 
-            var expectedSet = new HashSet<T>(elems);
-            expectedSet.ExceptWith(e);
+        /// <summary>
+        /// Execute the async chain.
+        /// </summary>
+        /// <param name="asyncAction"></param>
+        /// <param name="cancellationToken"></param>
+        /// <exception cref="InvalidOperationException"></exception>
+        public async Task ExecuteAsync(Func<Task>? asyncAction = null, CancellationToken cancellationToken = default)
```

Comment:
Calling this method will kick off the async execution. The async codes in the chain will be awaited one at a time and the executed flag will be set.
An async chain can only be executed once as a safety net because we can not guarantee TestSubscriber state.

Reply:
Got it and the `_executed` flag is what blows up bad configurations where this is called multiple times